posthog/src/main/java/com/posthog/PostHogConfig.kt

```diff
@@ -0,0 +1,149 @@
+package com.posthog
+
+import com.posthog.internal.BuildConfig
+import com.posthog.internal.PostHogContext
+import com.posthog.internal.PostHogLogger
+import com.posthog.internal.PostHogNetworkStatus
+import com.posthog.internal.PostHogPreferences
+import com.posthog.internal.PostHogPrintLogger
+
+/**
+ * The SDK Config
+ */
+public open class PostHogConfig(
+    /**
+     * The PostHog API Key
+     */
+    public val apiKey: String,
+
+    /**
+     * The PostHog Host
+     * Defaults to https://app.posthog.com
+     */
+    public val host: String = "https://app.posthog.com",
+
+    /**
+     * Logs the debug logs to the [logger] if enabled
+     * Defaults to false
+     */
+    public var debug: Boolean = false,
+
+    /**
+     * This flag prevents capturing any data if enabled
+     * Defaults to false
+     */
+    @Volatile
+    public var optOut: Boolean = false,
```

Comment:
thinking myself into implementer mindset... "Can I change opt out or is it fixed on startup? So could I use this to enable a user aware tracking flow?"

Reply:
You can change during init or using `PostHog.optIn()`/`PostHog.optOut()`

Reply:
That's the kind of info I think fits really well in comments like this that will surface in the IDE when a user if poking around to understand the SDK.

It _might_ be better just in the documentation with a link out though so you're not editing in multiple places

Reply:
Good idea, I will add to the dokka comment a well.